airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -91,6 +92,8 @@ const ReplicationView: React.FC<IProps> = ({ onAfterSaveSchema, connectionId })
     if (activeUpdatingSchemaMode) {
       setActiveUpdatingSchemaMode(false);
     }
+
+    formikHelpers?.resetForm({ values });
```

Comment:
This `resetForm({ values })` updates the Formik form with the new base values so that the form is no longer marked as "dirty" during an update. The confirmation modal works based on the dirtiness of the Formik form.

Here's the sequence of events:

Form is initialized with loaded values
User edits Form, and the form "dirty"
User submits form. Values are updated successfully
**Form is reset with updated values, thus longer "dirty"**